File: SDKs/amazonaws.com/kms/2014-11-01/go/pkg/models/operations/verify.go
package operations

import (
	"openapi/pkg/models/shared"
)

type VerifyXAmzTargetEnum string

const (
	VerifyXAmzTargetEnumTrentServiceVerify VerifyXAmzTargetEnum = "TrentService.Verify"
)

type VerifyHeaders struct {
	XAmzAlgorithm     *string              `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
	XAmzContentSha256 *string              `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
	XAmzCredential    *string              `header:"style=simple,explode=false,name=X-Amz-Credential"`
	XAmzDate          *string              `header:"style=simple,explode=false,name=X-Amz-Date"`
	XAmzSecurityToken *string              `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
	XAmzSignature     *string              `header:"style=simple,explode=false,name=X-Amz-Signature"`
	XAmzSignedHeaders *string              `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
	XAmzTarget        VerifyXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
}

type VerifyRequest struct {
	Headers VerifyHeaders
	Request shared.VerifyRequest `request:"mediaType=application/json"`
}

type VerifyResponse struct {
	ContentType                  string
	DependencyTimeoutException   *interface{}
	DisabledException            *interface{}
	InvalidGrantTokenException   *interface{}
	InvalidKeyUsageException     *interface{}
	KmsInternalException         *interface{}
	KmsInvalidSignatureException *interface{}
	KmsInvalidStateException     *interface{}
	KeyUnavailableException      *interface{}
	NotFoundException            *interface{}
	StatusCode                   int64
	VerifyResponse               *shared.VerifyResponse
}

File: SDKs/amazonaws.com/kms/2014-11-01/typescript/src/sdk/models/operations/verify.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";


export enum VerifyXAmzTargetEnum {
    TrentServiceVerify = "TrentService.Verify"
}


export class VerifyHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: VerifyXAmzTargetEnum;
}


export class VerifyRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  headers: VerifyHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.VerifyRequest;
}


export class VerifyResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  dependencyTimeoutException?: any;

  @SpeakeasyMetadata()
  disabledException?: any;

  @SpeakeasyMetadata()
  invalidGrantTokenException?: any;

  @SpeakeasyMetadata()
  invalidKeyUsageException?: any;

  @SpeakeasyMetadata()
  kmsInternalException?: any;

  @SpeakeasyMetadata()
  kmsInvalidSignatureException?: any;

  @SpeakeasyMetadata()
  kmsInvalidStateException?: any;

  @SpeakeasyMetadata()
  keyUnavailableException?: any;

  @SpeakeasyMetadata()
  notFoundException?: any;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  verifyResponse?: shared.VerifyResponse;
}
